collector/lib/Logging.h

```diff
@@ -71,25 +77,79 @@ class LogMessage {
               << std::left << std::setw(LevelPaddingWidth) << GetLogLevelName(level_)
               << " " << std::put_time(nowTm, "%Y/%m/%d %H:%M:%S")
               << "] ";
+  }
 
-    if (throttled_) {
-      std::cerr << "[Throttled] ";
-    }
-
-    if (include_file_) {
+  void PrintFile() {
+    if (CheckLogLevel(LogLevel::DEBUG)) {
       const char* basename = strrchr(file_, '/');
-      if (!basename) {
+      if (basename == nullptr) {
         basename = file_;
       } else {
         ++basename;
       }
       std::cerr << "(" << basename << ":" << line_ << ") ";
     }
+  }
+
+ private:
+  const char* file_;
+  int line_;
+  LogLevel level_;
+};
+
+class LogHeader : public ILogHeader {
```

Comment:
Future us would appreciate a few lines of comments, explaining what's the difference between this and the following classes.

Reply:
Makes sense, I'll document them.